### ISSUE TITLE
magnet links do not work under win10

### ISSUE THREAD

**Issue by nickolasclarke** [kzahel/jstorrent-legacy-app#136]
when I enable magnet support, I get a popup from win10 staying, "To change your default apps, go to settings > default apps > app". When I look through the options of "choose default by protocol" there is no option for magnetic links. When I try to force the change by going to `chome://settings/handlers` and manually set the `site` value to `jstorrent.com` in the dropdown, the same popup appears, and then it resets the setting in chrome to `none`

**Comment by kzahel** [comment]
Thanks for the report.
Wow, that was annoying.
I luckily had a spare Win10 VM lying around, and was able to make it work. The Setup button seemed to not work, but it just didn't focus Chrome.
 I saw the same dialog pop up, and I went to settings > default apps

I let Google Chrome get access to magnet and hit OK, but it still did not seem to work. So I gave chrome access to be default handler for everything (mailto, https, etc)

Then went to chrome://settings/handlers and selected jstorrent to handle magnet

then it worked

**Comment by kzahel** [comment]
I don't see an open issue here https://bugs.chromium.org/p/chromium/issues/list?can=2&q=windows+10+protocol+handler&colspec=ID+Pri+M+Stars+ReleaseBlock+Component+Status+Owner+Summary+OS+Modified&x=m&y=releaseblock&cells=ids

**Comment by kzahel** [comment]
https://twitter.com/ericlaw/status/626939313977098240

**Comment by kzahel** [comment]
https://bugs.chromium.org/p/chromium/issues/detail?id=511429#c5

**Comment by kzahel** [comment]
"
So it looks like the only way for a website to register as default protocol handler on Win 10 is to first (!) select Chrome as the default handler and THEN to select a webservice. Doing it the other way around will just throw the blue bar of uselessness from MS.
"

**Comment by nickolasclarke** [comment]
how do you add the magnet protocol to the list in the first place though? It does not show up for me in the wind10 default apps for protocol settings, and I've got all the usual web protocol suspects set to chrome already (http, https, ftp, mailto).

**Comment by kzahel** [comment]
Hmm, I don't know, it was just there in the list, right alongside http/ftp/mailto etc

**Comment by nickolasclarke** [comment]
humm, I'll make a more thorough search. if only they had a universal search box throughout their settings UI....

**Comment by nickolasclarke** [comment]
Ah, you need to use the "select defaults by app'" in lieu of choose defaults by protocol. The fomer loads up an old settings program "Set Program Associations" and there you can choose "set associations for a program" and then it shows up under protocols. Perhaps it would be good to add these instructions to the page that guides users through enabling mag links.

**Comment by kzahel** [comment]
Yeah on the left there was a list of apps and I scrolled to Google Chrome and gave chrome access to everything. I never looked for or saw the other dialog you mention.

I'll add a link to the page if I see the user agent is windows 10, and let them know they're in for some pain :-)

**Comment by kzahel** [comment]
The legacy Chrome App is no longer maintained. I've rebuilt JSTorrent from scratch — the new version has a proper desktop app for Windows with magnet link handling built in. Get it at **[jstorrent.com](https://jstorrent.com)**

If you run into any issues, please leave feedback!
